onnxruntime/test/providers/cpu/nn/conv_transpose_op_test.cc

```diff
@@ -533,5 +532,25 @@ TEST(ConvTransposeTest, ConvTranspose_DefaultStridesAndDilations) {
   TestConvTransposeOp(attrs, {X, W}, {X_shape, W_shape}, expected_vals, Y_shape);
 }
 
+TEST(ConvTransposeTest, ConvTranspose_1D_NonDefaultStridesAndDilations) {
+  ConvTransposeOpAttributes attrs = {
```

Comment:
This combination of values triggers incorrect shape to be calculated. With the exisitng logic output shape will be {1, 1, 8} which is incorrect. The right shape is {1, 1, 12}. This aligns with ONNX shape inference (verified by making all inputs initializers to trigger shape inferencing)

Reply:
Since this new test case is under `providers/cpu/`, I presume it only tests the CPU provider? If so, that's sad it would be nice to validate this bug against other providers too, like DirectML :). Ideally, I think whenever we fix a bug, it would be nice to add a test mini-model into the ONNX backend. 

On a related note, WinML's ONNX conformance tests execute both CPU and GPU. I should check with Brian Martin to see if/when WinML is planning to migrate those 2'000 additional test cases over into GitHub.

Reply:
I think the other EPs are tested too. EPs will have to be explicitly removed for the unit test.

For this op, only TensorRT is being excluded for a reson as stated in the comment - 

  test.Run(expect_result, err_str, excluded_provider_types);  // Disable TensorRT because weight as input is not supported